extension/src/state_aggregate.rs

```diff
@@ -2272,4 +2273,95 @@ SELECT toolkit_experimental.duration_in('one', toolkit_experimental.compact_stat
+            // only a single entry so ordering is consistent between runs
+            client
+                .update(
+                    r#"INSERT INTO states VALUES
+                        ('2020-1-1 10:00', 'starting');
+                    INSERT INTO states_int VALUES
+                        ('2020-1-1 10:00', -67876545);"#,
+                    None,
+                    None,
+                )
+                .unwrap();
+
+            let agg_text = select_one!(
+                client,
+                "SELECT state_agg(ts, state)::TEXT FROM states",
+                &str
+            );
+            let expected = "(version:1,compact_state_agg:(version:1,states_len:8,durations_len:1,durations:[(duration:0,state:(a:0,b:8))],combined_durations_len:1,combined_durations:[(start_time:631188000000000,end_time:631188000000000,state:(a:0,b:8))],first_time:631188000000000,last_time:631188000000000,first_state:0,last_state:0,states:[115,116,97,114,116,105,110,103],compact:false,integer_states:false))";
+            assert_eq!(agg_text, expected);
+
+            let agg_text = select_one!(
+                client,
+                "SELECT state_agg(ts, state)::TEXT FROM states_int",
+                &str
+            );
+            let expected = "(version:1,compact_state_agg:(version:1,states_len:0,durations_len:1,durations:[(duration:0,state:(a:9223372036854775807,b:-67876545))],combined_durations_len:1,combined_durations:[(start_time:631188000000000,end_time:631188000000000,state:(a:9223372036854775807,b:-67876545))],first_time:631188000000000,last_time:631188000000000,first_state:0,last_state:0,states:[],compact:false,integer_states:true))";
+            assert_eq!(agg_text, expected);
+        });
+    }
+
+    #[pg_test]
+    fn combine() {
+        assert_eq!(state_agg::combine(None, None), None);
+
+        let mut trans_state_2 = CompactStateAggTransState::new(true);
+        trans_state_2.record(MaterializedState::Integer(444), 10005000);
+        let mut trans_state_1 = CompactStateAggTransState::new(true);
+        trans_state_1.record(MaterializedState::Integer(333), 10000000);
+        let trans_state = state_agg::combine(Some(&trans_state_1), Some(&trans_state_2)).unwrap();
+        let trans_state = state_agg::combine(Some(&trans_state), None).unwrap();
+        let trans_state = state_agg::combine(None, Some(&trans_state)).unwrap();
+        assert_eq!(
+            trans_state,
+            CompactStateAggTransState {
+                records: vec![
+                    Record {
+                        state: MaterializedState::Integer(333),
+                        time: 10000000
+                    },
+                    Record {
+                        state: MaterializedState::Integer(444),
+                        time: 10005000
+                    }
+                ],
+                integer_states: true,
+            }
+        );
+    }
+
+    #[pg_test]
+    fn binary_serialization() {
```

Comment:
I've made this mistake in a few places as well, but we should be testing the binary state of the finalized aggregate, not the trans state.  The trans state's binary state will never persist into a version update, so we shouldn't need to test it.